python_dwd/data_storing.py

```diff
@@ -0,0 +1,120 @@
+""" Data storing/restoring methods"""
+from pathlib import Path
+from typing import Tuple, Union
+import pandas as pd
+
+from python_dwd.additionals.helpers import create_stationdata_dtype_mapping
+from python_dwd.constants.metadata import STATIONDATA_NAME, H5_FORMAT
+from python_dwd.enumerations.parameter_enumeration import Parameter
+from python_dwd.enumerations.period_type_enumeration import PeriodType
+from python_dwd.enumerations.time_resolution_enumeration import TimeResolution
+
+
+def store_dwd_data(station_data: pd.DataFrame,
+                   station_id: int,
+                   parameter: Parameter,
+                   time_resolution: TimeResolution,
+                   period_type: PeriodType,
+                   folder: str) -> None:
+    """
+    Function to store data in a local file hdf file. The function takes a pandas
+    DataFrame plus additionally the request parameters to identify data within the
+    hdf file and another folder argument for the place where the file is stored.
+
+    Args:
+        station_data: the pandas DataFrame with the obtained data
+        station_id: the id of the station of which the data is stored in the DataFrame
+        parameter: the parameter enumeration
+        time_resolution: the time resolution enumeration
+        period_type: the period type as enumeration
+        folder: the folder where the hdf is stored
+    Returns:
+        None, prints information if data was not stored
+    """
+    # Make sure that there is data that can be stored
+    if station_data.empty:
+        print("There is no data in 'station_data' that can be stored.")
+
+        return
+
+    request_string = _build_local_store_key(
+        station_id, parameter, time_resolution, period_type)
+
+    local_filepath = Path(folder, STATIONDATA_NAME).absolute() / \
```

Comment:
Refactor path generation as a function and make a test for it. Then use this where you want to get access to such a data.